src/ray/core_worker/core_worker.h

```diff
@@ -1051,6 +1051,17 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
   std::unordered_map<std::string, std::vector<uint64_t>> GetActorCallStats() const;
 
  private:
+  void BuildCommonTaskSpec(
+      TaskSpecBuilder &builder, const JobID &job_id, const TaskID &task_id,
+      const std::string &name, const TaskID &current_task_id, const uint64_t task_index,
+      const TaskID &caller_id, const rpc::Address &address, const RayFunction &function,
+      const std::vector<std::unique_ptr<TaskArg>> &args, uint64_t num_returns,
+      const std::unordered_map<std::string, double> &required_resources,
+      const std::unordered_map<std::string, double> &required_placement_resources,
+      const BundleID &bundle_id, bool placement_group_capture_child_tasks,
+      const std::string &debugger_breakpoint, const std::string &serialized_runtime_env,
+      const std::vector<std::string> &runtime_env_uris,
+      const std::string &concurrency_group_name = "");
```

Comment:
hmm I'm confused wasn't this already here? did you make it private? if so, remove the public one?

Reply:
I moved this function from a static function(only defined in "core_worker.cc") to a member function of `CoreWorker` class. This function is only used by `CoreWorker`. And the original reason is that I need to use `job_config_` to write runtime_env in this function. `job_config_` is a member of `CoreWorker`.

Reply:
Gotcha, makes sense!